### DISCUSSION TITLE
v1.1.1

### DISCUSSION THREAD

**Discussion by methane** [PyMySQL/PyMySQL#1171] [General]
> [!WARNING]
> This release fixes a vulnerability (CVE-2024-36039).
> All users are recommended to update to this version.
>
> If you can not update soon, check the input value from untrusted source has an expected type.
> Only dict input from untrusted source can be an attack vector.

## What's Changed

* Prohibit dict parameter for `Cursor.execute()`. It didn't produce valid SQL
  and might cause SQL injection. (CVE-2024-36039)

## Merged PRs

* Add support for Python 3.12 by @hugovk in https://github.com/PyMySQL/PyMySQL/pull/1134
* chore(deps): update actions/checkout action to v4 by @renovate in https://github.com/PyMySQL/PyMySQL/pull/1136
* Update codecov/codecov-action action to v4 by @renovate in https://github.com/PyMySQL/PyMySQL/pull/1137
* ci: use codecov@v3 by @methane in https://github.com/PyMySQL/PyMySQL/pull/1142
* chore(deps): update dessant/lock-threads action to v5 by @renovate in https://github.com/PyMySQL/PyMySQL/pull/1141
* doc: use rtd theme by @methane in https://github.com/PyMySQL/PyMySQL/pull/1143
* use Ruff as formatter by @methane in https://github.com/PyMySQL/PyMySQL/pull/1144
* Added ssl_key_password param by @svaskov in https://github.com/PyMySQL/PyMySQL/pull/1145
* chore(deps): update dependency sphinx-rtd-theme to v2 by @renovate in https://github.com/PyMySQL/PyMySQL/pull/1147
* chore(deps): update actions/setup-python action to v5 by @renovate in https://github.com/PyMySQL/PyMySQL/pull/1152
* chore(deps): update github/codeql-action action to v3 by @renovate in https://github.com/PyMySQL/PyMySQL/pull/1154
* chore(deps): update codecov/codecov-action action to v4 by @renovate in https://github.com/PyMySQL/PyMySQL/pull/1158
* Support error packet without sqlstate by @methane in https://github.com/PyMySQL/PyMySQL/pull/1160
* test json - mariadb without JSON type by @grooverdan in https://github.com/PyMySQL/PyMySQL/pull/1165

## New Contributors
* @hugovk made their first contribution in https://github.com/PyMySQL/PyMySQL/pull/1134
* @svaskov made their first contribution in https://github.com/PyMySQL/PyMySQL/pull/1145

**Full Changelog**: https://github.com/PyMySQL/PyMySQL/compare/v1.1.0...v1.1.1

<hr /><em>This discussion was created from the release <a href='https://github.com/PyMySQL/PyMySQL/releases/tag/v1.1.1'>v1.1.1</a>.</em>

**Comment by cristopherchacon** [comment]
Why is this change not considered a breaking change?

**Reply by methane** [reply]
Because there were no valid usage for this function.

**Reply by methane** [reply]
See this thread.

https://github.com/PyMySQL/PyMySQL/issues/1172#issuecomment-2124750425
